homework-submission/todo-app/src/stores/TodoStore.js

```diff
@@ -0,0 +1,137 @@
+import { action, observable, computed } from 'mobx';
+import todoList from '../components/todos.json';
+import moment from 'moment';
+import '../App.css';
+
+class TodoStore {
+
+    @observable
+    listTodo = todoList
+
+    @observable
+    emptyListTodo = {}
+
+    @observable
+    defaultValue = {
+        id: '',
+        description: '',
+        deadline: moment(),
+        done: false
+    }
+
+    @observable
+    editing = false
+
+
+    @computed
+    get completedTodosCount() {
+        return this.listTodo.filter(
+            todo => todo.done === true
+        ).length;
+    }
+    @computed
+    get todosCount() {
+        return this.listTodo.length;
+    }
+
+
+
+    @action
+    handleCheckBox = (id) => {
+        const newTodoList = this.listTodo.map((todoElement) => {
+            if (todoElement.id === id) {
+
+                return {
+                    ...todoElement,
+                    done: !todoElement.done
+                }
+
+            }
+            else {
+                return todoElement;
+            }
+
+        });
+
+        this.listTodo = newTodoList;
+
+    }
+
+    @action
+    onChanging = (index, value) => {
+        const newValue = this.defaultValue;
+        newValue[index] = value;
+        this.defaultValue = newValue;
+
+    };
+
+    @action
+    onSubmitAdd = (e) => {
+        e.preventDefault();
+        if (e.target.description.value === '' || e.target.deadline.value === '') {
+            alert('You must Enter Description and Date!!')
+            return null;
+        }
+        else {
+
+            let stateTodos = this.listTodo;
```

Comment:
here should be let :)

Reply:
because this variable is firing when i click add todo and every props changes .

Reply:
No Actually any thing that is an Object or Array or Function(_if used_) it's a best practice to use `const` 
because it takes a static memory.
so you can but it like so and it will work fine:
```javascript
 const stateTodos = {
        id: this.listTodo.length + 1,
        description: e.target.description.value,
        deadline: e.target.deadline.value,
        done: e.done
};
```
same as yours, but you are just adding an extra reference.